.github/workflows/memory-check.yml

```
@@ -0,0 +1,157 @@
+              print("::warning::High memory usage detected (>500MB)")
+              sys.exit(1)
+          
+          sys.exit(exit_code)
+          EOF
+          
+          python profile_memory.py
+      
+      - name: Check for memory leaks in Python
+        run: |
+          source .venv/bin/activate
+          
```

YAMLlint (1.37.1)

[error] trailing spaces

(trailing-spaces)